### DISCUSSION TITLE
ORF (Olypus Raw File) images have wrong colors (too red)

### DISCUSSION THREAD

**Discussion by LuckyLuke-n** [dlemstra/Magick.NET#1264] [Help] [answered]
Hi all,
I need your help with a color issue.  After opening with Magick.NET, .orf-files (Olympus Raw Format) appear to have much more red color than they originally have. Below I present you an example.

| After opening with Magick.NET | Original image |
| -------------------------------- | -------------- |
| ![grafik](https://user-images.githubusercontent.com/67018627/192777622-a35637df-2d24-4f1b-8b94-a0f09fdc7407.png) | ![grafik](https://user-images.githubusercontent.com/67018627/192777685-9e7e8251-0526-4db3-a261-fda0ddfb4d0e.png) |

I have tried this with different formats:
| Format | Colors correct? |
| ------- | ----------- |
| ORF (Olypus Raw Format) | ❌ |
| JPG | ✔️ |
| TIFF | ✔️ |

I do this to open the .orf-file and convert it to a bitmap file:
```
MagickImage magickImage = new(pathToFile);
bitmap = magickImage.ToBitmap();
```

The same happens when I only save the orf-file as a jpg-file:
```
MagickImage magickImage = new(pathToFile);
magickImage.Write(filename, MagickFormat.Jpg);
```
I am very much looking forward to your feedback. Huge thanks in advance.

**Comment by dlemstra** [comment, marked as answer]
For the `ORF` file you probably want to play with the `DngReadDefines` and adjust some of the settings to get the output that you want.

**Reply by LuckyLuke-n** [reply]
Thank you very much. This fixed my issue.

In case someone has a similar problem, that is what I did:
```
// color correction for orf files
DngReadDefines dngReadDefines = new()
{
    OutputColor = DngOutputColor.SRGB, // default
    UseCameraWhitebalance = true // this removes the dreadful red/brown color
};
MagickReadSettings readSettings = new(dngReadDefines);

MagickImage magickImage = new(filename, readSettings);
magickImage.BrightnessContrast((Percentage)0, (Percentage)5, Channels.Red); // this corrects the missing red after correcting the white balance
// magickImage.Sharpen(); // takes quite a long time and would slightly improve the image quality
```
